### PR TITLE
fix history upsert tx

### LINKED ISSUE #219: upserted entites in the history use the wrong tx ID

### PR DESCRIPTION
**Pull request by kordano**
closes #219

### CODE REVIEW

File: test/datahike/test/time_variance.cljc [hunk 1]
```diff
@@ -155,3 +156,60 @@
...
+      (d/transact conn [[:db/add [:name "Alice"] :age 35]])
+      (is (= #{[35 (+ const/tx0 3) true]}
+             (d/q query @conn)))
+      (is (= #{[25 (+ const/tx0 1) true]
+               [25 (+ const/tx0 2) false]
+               [30 (+ const/tx0 2) true]
+               [30 (+ const/tx0 3) false]
+               [35 (+ const/tx0 3) true]}
+             (d/q query (d/history @conn)))))
+    (testing "re-insert previous value"
```
[the remark this answers is not in the document]
Comment (kordano) [reply test/datahike/test/time_variance.cljc:191]: It's the same that was added earlier. So 25 was the original value.

File: src/datahike/db.cljc [hunk 2]
```diff
@@ -1238,7 +1241,7 @@
         (if (= (.-v old-datom) v)
           report
           (-> report
-              (transact-report (datom e a (.-v old-datom) (if keep-history? (datom-tx old-datom) tx) false))
+              (transact-report (datom e a (.-v old-datom) tx false))
```
[the remark this answers is not in the document]
Comment (kordano) [reply src/datahike/db.cljc:1244]: Yes.

**Comment by kordano** [comment]
I merged the latest development branch in here. Should be ready for review again.

File: src/datahike/db.cljc [hunk 3]
```diff
@@ -1120,12 +1124,11 @@
                 (and keep-history? indexing?) (update-in [:temporal-avet] #(di/-insert % datom :avet)))
         db))))
 
-
 (defn- with-temporal-datom [db ^Datom datom]
   (let [indexing? (indexing? db (.-a datom))
         schema? (ds/schema-attr? (.-a datom))
         current-datom ^Datom (first (-search db [(.-e datom) (.-a datom) (.-v datom)]))
-        history-datom ^Datom (first (search-temporal-indices db [(.-e datom) (.-a datom) (.-v datom)]))
+        history-datom ^Datom (first (search-temporal-indices db [(.-e datom) (.-a datom) (.-v datom) (.-tx datom)]))
```
Comment (jsmassa) [inline src/datahike/db.cljc:1131]: So happened before you made this change?

**Review by jsmassa — APPROVED** [review]
Looks good to me

File: src/datahike/db.cljc (hunk 3 above)
Comment (kordano) [reply src/datahike/db.cljc:1131]: Right, the tx id in the removed datom was not the tx id from the when the removal happened but the original insertion tx id.

Comment (jsmassa) [reply src/datahike/db.cljc:1131]: I see, thanks

**Pull request merged**